src/Swift.Runtime/src/Metadata/TypeMetadata.cs

```diff
@@ -0,0 +1,199 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.Collections.Generic;
+
+namespace Swift.Runtime;
+
+/// <summary>
+/// Flags used to describe types
+/// </summary>
+[Flags]
+public enum TypeMetadataFlags {
+    /// <summary>
+    /// The metadata is not an actual type
+    /// </summary>
+    MetadataKindIsNonType = 0x400,
+    /// <summary>
+    /// The metadata doesn't live on the heap
+    /// </summary>
+    MetadataKindIsNonHeap = 0x200,
+    /// <summary>
+    /// The type is private to the runtime
+    /// </summary>
+    MetadataKindIsRuntimePrivate = 0x100,
```

Comment:
Is `MetadataKind` a common prefix for all values, or will there be more flags that aren't `MetadataKind` flags? If the former, the prefix should be removed:

```suggestion
    IsNonType = 0x400,
    /// <summary>
    /// The metadata doesn't live on the heap
    /// </summary>
    IsNonHeap = 0x200,
    /// <summary>
    /// The type is private to the runtime
    /// </summary>
    IsRuntimePrivate = 0x100,
```